drivers/SmartThings/matter-switch/src/init.lua

```diff
@@ -599,7 +611,8 @@ local function try_build_child_switch_profiles(driver, device, switch_eps, main_
           }
         )
         parent_child_device = true
-        if _ == 1 and child_profile == "light-power-energy-powerConsumption" then
+        if _ == 1 and
+           (child_profile == "light-power-energy-powerConsumption" or child_profile == "light-level-power-energy-powerConsumption") then
```

Comment:
let's generalize this as well for any future energy devices we'd like to profile like this:
```suggestion
if _ == 1 and string.find(child_profile, "energy") then 
```
now any child profile using with the name `energy` in it will use this handling if necessary

Reply:
Thanks, I applied it.